include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -339,3 +339,53 @@ def MuxOp : RTLOp<"mux",
     }]>
   ];
 }
+
+def BitArrayType : Type<CPred<[{
+  $_self.isa<ArrayType>() &&
+  $_self.cast<ArrayType>().getElementType().isSignlessInteger(1)
+}]>, "RTL array of bits">;
+
+def KnownBitWidthType : Type<CPred<[{getBitWidth($_self) != -1}]>,
+  "Type wherein the bitwidth in hardware is known">;
+
+def CastToBitsOp: RTLOp<"cast.tobits", [NoSideEffect]> {
+  let summary = "Cast an arbitrary value to a bit array";
+  let description = [{
+    Assign the bits from a value to a bitarray without any sort of
```

Comment:
This isn't just to a bitarray though, right?

Reply:
It is. I wanted to keep this initial PR very restrictive but I can certainly open it up.